connectors/source.py

```diff
@@ -304,6 +304,16 @@ def basic_rules_validators(cls):
             BasicRulesSetSemanticValidator,
         ]
 
+    @classmethod
```

Comment:
this is what I would like to propose:

```
class BaseDataSource:
    def get_id(self, doc):
        raise NotImplementedError

    def hash_id(self, doc):
        # cost to be measured
        return hashlib.md5(str(path).encode("utf8")).hexdigest()
```

and then always use it in the sync

Reply:
Also: we should keep the original id in a `_original_id` in the doc, just to make sure we can track down just in case